src/client/mesh_generator_thread.cpp

```diff
@@ -149,6 +149,7 @@ QueuedMeshUpdate *MeshUpdateQueue::pop()
 		m_queue.erase(i);
 		m_urgents.erase(q->p);
 		fillDataFromMapBlockCache(q);
+		q->urgent = must_be_urgent;
```

Comment:
shouldn't this be set at the point where's it put into the queue?

Reply:
Moved there.